File: android/app/src/main/java/com/emwaver/emwaverandroidapp/files/UserFileData.java
/*
 * EMWaver
 * Copyright (c) 2026 Luís Marnoto
 * All rights reserved.
 */

package com.emwaver.emwaverandroidapp.files;

public final class UserFileData {

    private final UserFileMetadata metadata;
    private final String textContent;
    private final byte[] binaryContent;

    public UserFileData(UserFileMetadata metadata, String textContent, byte[] binaryContent) {
        this.metadata = metadata;
        this.textContent = textContent;
        this.binaryContent = binaryContent;
    }

    public UserFileMetadata getMetadata() {
        return metadata;
    }

    public boolean hasTextContent() {
        return textContent != null;
    }

    public String getTextContent() {
        return textContent;
    }

    public boolean hasBinaryContent() {
        return binaryContent != null;
    }

    public byte[] getBinaryContent() {
        return binaryContent;
    }
}

File: apple/EMWaverAppleCore/Sources/EMWaverScriptStorage/UserFileData.swift
/*
 * EMWaver
 * Copyright (c) 2026 Luis Marnoto
 * SPDX-License-Identifier: Apache-2.0
 */

import Foundation

public struct UserFileData {
    public let metadata: UserFileMetadata
    public let textContent: String?
    public let binaryContent: Data?

    public var hasTextContent: Bool { textContent != nil }
    public var hasBinaryContent: Bool { binaryContent != nil }

    public init(metadata: UserFileMetadata, textContent: String?, binaryContent: Data?) {
        self.metadata = metadata
        self.textContent = textContent
        self.binaryContent = binaryContent
    }
}
